### PR TITLE
Make CONTRIBUTING.rst be up to date.

### LINKED ISSUE #3205: CONTRIBUTING.rst out of date since the switch to nox from tox.
Since we're using nox now, we should update the CONTRIBUTING.rst to show users how to run docs/lint/unit/system tests.

### PR DESCRIPTION
**Pull request by lukesneeringer**
Fixes #3205.

### CODE REVIEW

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -21,7 +21,7 @@ In order to add a feature to ``google-cloud-python``:
   documentation (in ``docs/``).
 
 - The feature must work fully on the following CPython versions:  2.7,
-  3.4, and 3.5 on both UNIX and Windows.
+  3.4, 3.5, and 3.6 on both UNIX and Windows.
```
Comment (dhermes) [inline CONTRIBUTING.rst]: ASIDE: Let's stop using `3.4`

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -68,11 +68,15 @@ Using a custom ``virtualenv``
...
 
   You can choose which Python version you want to use by passing a ``--python``
-  flag to ``virtualenv``.  For example, ``virtualenv --python python2.7``
-  chooses the Python 2.7 interpreter to be installed.
+  flag to ``virtualenv``.  For example, ``virtualenv --python python3.6``
+  chooses the Python 3.6 interpreter to be installed.
+
+  .. note::
+      We recommend developing in Python 3, and using the test suite to
+      ensure compatibility with Python 2.
```
Comment (dhermes) [inline CONTRIBUTING.rst]: :+1:

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -91,43 +95,32 @@ Using a custom ``virtualenv``
...
-    $ tox -e py27
-    $ tox -e py34
-    $ ...
-
-- If you'd like to poke around your code in an interpreter, let
-  ``tox`` install the environment of your choice::
-
-    $ # Install only; without running tests
-    $ tox -e ${ENV} --recreate --notest
+We use `nox`_ to instrument our tests.
```
Comment (dhermes) [inline CONTRIBUTING.rst]: This is invalid, not sure why:

screenshot from 2017-08-08 09-57-07

File: CONTRIBUTING.rst [hunk 4]
```diff
@@ -91,43 +95,32 @@ Using a custom ``virtualenv``
...
   currently developing and run tests there::
 
     $ export GIT_ROOT=$(pwd)
-    $ cd ${GIT_ROOT}/core/
-    $ tox -e py27
     $ cd ${GIT_ROOT}/datastore/
-    $ tox -e py27
+    $ nox -s "unit_tests(python_version='3.6')"
+
+.. nox: https://pypi.org/project/nox-automation/
```
Comment (dhermes) [inline CONTRIBUTING.rst]: Ah I see, you want `.. _nox: https://...` (i.e. you are missing a `_`)

File: CONTRIBUTING.rst [hunk 5]
```diff
@@ -270,90 +234,21 @@ Running System Tests
...
-   $ python system_tests/populate_datastore.py
+   $ python datastore/tests/system/utils/populate_datastore.py
 
 - If you make a mistake during development (i.e. a failing test that
   prevents clean-up) you can clear all system test data from your
   datastore instance via::
 
-   $ python system_tests/clear_datastore.py
-
-System Test Emulators
```
Comment (dhermes) [inline CONTRIBUTING.rst]: @lukesneeringer Why did you delete this? Running in the emulators is very useful.

File: CONTRIBUTING.rst (hunk 1 above)
Comment (tseaver) [inline CONTRIBUTING.rst]: FWIW, The 3.4 release line is in ["security fixes only" mode](http://permalink.gmane.org/gmane.comp.python.committers/4809), and not sunsetted (as 3.3 is about to be).
